RadixWallet/Clients/OnLedgerEntitiesClient/Helpers/OnLedgerEntitiesClient+ComplexTypes.swift

```diff
@@ -72,24 +93,27 @@ extension OnLedgerEntitiesClient {
 		// Normal fungible resource
 		let isXRD = resourceAddress.isXRD(on: networkID)
 		let isHidden = hiddenResources.contains(.fungible(resourceAddress))
-		let details: ResourceBalance.Fungible = .init(isXRD: isXRD, amount: .init(nominalAmount: amount), guarantee: guarantee)
+		let details: KnownResourceBalance.Fungible = .init(isXRD: isXRD, amount: resourceAmount, guarantee: guarantee)
 
 		return .init(resource: resource, details: .fungible(details), isHidden: isHidden)
 	}
 
 	private func poolUnit(
 		_ resource: OnLedgerEntity.Resource,
-		amount: Decimal192,
+		amount: ResourceAmount,
 		poolContributions: [some TrackedPoolInteraction] = [],
 		entities: InteractionReview.Sections.ResourcesInfo = [:],
 		resourceAssociatedDapps: InteractionReview.Sections.ResourceAssociatedDapps? = nil,
 		networkID: NetworkID,
 		guarantee: TransactionGuarantee?,
 		hiddenResources: [ResourceIdentifier]
-	) async throws -> ResourceBalance {
+	) async throws -> KnownResourceBalance {
 		let resourceAddress = resource.resourceAddress
 
 		if let poolContribution = poolContributions.first(where: { $0.poolUnitsResourceAddress == resourceAddress }) {
+			guard let amount = amount.exactAmount?.nominalAmount else {
+				fatalError("Not possible")
```

Comment:
Add a better message as to why this is not possible.
It is not possible because the poolContributions can happen only with an enclosed manifest, where the amounts are exact.

Still, it may be worth investigating if we could have separate functions that are called when resolving the sections for manifestSummary or from executionSummary.
For the case of manifestSummary, there cannot be `poolContributions`.

Reply:
Did the separation [here](https://github.com/radixdlt/babylon-wallet-ios/pull/1377/commits/9ad60a00f7de3a6ad41dae88b9ab35a90c0ccd2c)